src/components/ProductPage/ProductMetaData.tsx

```diff
@@ -73,10 +72,7 @@ const ProductMetaData: React.FC<ProductMetaDataProps> = ({ tokenData }) => {
         <StyledStat>
           <StyledStatTitle>Current Supply</StyledStatTitle>
           <StyledStatMetric>
-            {formatCurrentSupply(
-              tokenData.latestMarketCap,
-              tokenData.latestPrice
-            )}
+            {formattedCurrentySupply !== '0' ? formattedCurrentySupply : '--'}
```

Comment:
This could be better. Open to suggestions

Reply:
Perhaps put that logic in the formatter, then this is a pure function read

```
const formatCurrentSupply = (currentSupply: BigNumber | undefined) =>
    currentSupply ? numeral(currentSupply).format('0,0')  :  '--'
```